src/compiler/utilities.ts

```diff
@@ -6104,7 +6104,14 @@ namespace ts {
             getSymlinkedFiles: () => symlinkedFiles,
             getSymlinkedDirectories: () => symlinkedDirectories,
             setSymlinkedFile: (path, real) => (symlinkedFiles || (symlinkedFiles = new Map())).set(path, real),
-            setSymlinkedDirectory: (path, directory) => (symlinkedDirectories || (symlinkedDirectories = new Map())).set(path, directory),
+            setSymlinkedDirectory: (path, directory) => {

```

Comment:
This should be handled when it is used for module specifier resolution instead of in general in my opinion since we still want to resolve the cache for answering if file is from referenced project for project construction ?

Reply:
I think something inside `node_modules/.pnpm/` should never be a project reference redirect, which seems to be the only reason beside module specifier resolution that this is used (`fileOrDirectoryExistsUsingSource`).

Reply:
You are right because it would resolve completely to actual path even if equivalent of npm link kind of thing is used .